FxAClient/Frontend/SignIn/FxASignInWebViewController.swift

```diff
@@ -0,0 +1,179 @@
+        let source = getJS()
+        let userScript = WKUserScript(
+            source: source,
+            injectionTime: WKUserScriptInjectionTime.AtDocumentEnd,
+            forMainFrameOnly: true
+        )
+
+        // Handle messages from the content server (via our user script).
+        let contentController = WKUserContentController()
+        contentController.addUserScript(userScript)
+        contentController.addScriptMessageHandler(
+            self,
+            name: "accountsCommandHandler"
+        )
+
+        let config = WKWebViewConfiguration()
+        config.userContentController = contentController
+
+        webView = WKWebView(
+            frame: CGRectMake(0, 0, 0, 0),
+            configuration: config
+        )
+        webView.navigationDelegate = navigationDelegate
+        if let url = url {
+            webView!.loadRequest(NSURLRequest(URL: url))
+        }
+
+        // Don't allow overscrolling.
+        webView.scrollView.bounces = false
+    }
+
+    func didCancel() {
+        delegate?.didCancel()
+    }
+
+    // Send a message to the content server.
+    func injectData(type: String, content: [String: AnyObject]) {
+        let data = [
+            "type": type,
+            "content": content,
+        ]
+        let json = JSON(data).toString(pretty: false)
+        let script = "window.postMessage(\(json), '\(url)');"
+        NSLog("reponse: \(json)")
+        webView!.evaluateJavaScript(script, completionHandler: nil)
+    }
+
+    private func onCanLinkAccount(data: JSON) {
+        //    // We need to confirm a relink - see shouldAllowRelink for more
+        //    let ok = shouldAllowRelink(accountData.email);
+        let ok = true
+        injectData("message", content: ["status": "can_link_account", "data": ["ok": ok]]);
+    }
+
+    // We're not signed in to a Firefox Account at this time, which we signal by returning an error.
+    private func onSessionStatus(data: JSON) {
+        injectData("message", content: ["status": "error"])
+    }
+
+    // We're not signed in to a Firefox Account at this time. We should never get a sign out message!
+    private func onSignOut(data: JSON) {
+        injectData("message", content: ["status": "error"])
+    }
+
+    // The user has signed in to a Firefox Account.  We're done!
+    private func onLogin(data: JSON) {
+        NSLog("Logged in with email: %@ and uid: %@", data["email"].asString!, data["uid"].asString!)
+        injectData("message", content: ["status": "login"])
+        dismissViewControllerAnimated(true, completion: nil)
+        self.delegate?.didSignIn(data)
+    }
+
+    // The content server page is ready to be shown.
+    private func onLoad(data: JSON) {
+        delegate?.didLoad()
+    }
+
+    // Handle a message coming from the content server.
+    func handleRemoteCommand(command: String, data: JSON) {
+        NSLog("command: %@", command)
+        switch command {
```

Comment:
I've implemented this, although with the `if let` and subsequent `switch` it's not really a win.